subprojects/plugins/src/main/java/org/gradle/api/plugins/jvm/internal/DefaultJvmTestSuite.java

```diff
@@ -120,6 +133,8 @@ public DefaultJvmTestSuite(String name, DependencyFactory dependencyFactory, Con
         Configuration runtimeOnly = configurations.getByName(sourceSet.getRuntimeOnlyConfigurationName());
         Configuration annotationProcessor = configurations.getByName(sourceSet.getAnnotationProcessorConfigurationName());
 
+        runtimeClasspath = configurations.getByName(sourceSets.getByName(SourceSet.MAIN_SOURCE_SET_NAME).getRuntimeClasspathConfigurationName());
```

Comment:
Perhaps we can move this logic into `useSpockForDerivedGroovyVersion` and verify the main source set exists before trying to get the `runtimeClasspath` -- for the case where the `JavaPlugin` isn't applied. 

What worries me this that this _kinda_ goes against the idea of having our tests depend on the production code via dependency-management. This PR adds a new dependency on the non-dependency management path by making it so any test which uses Spock now directly interacts with the production code via its configurations instead of through dependency-management. 

Granted, the more "dependency-management-style" solution would probably involve us creating our own detached configuration and adding a `project()` dependency to it, then resolving that and checking for the Groovy jar. I'm not saying we should do this in this PR or at all, but I can kinda see why some integ tests are failing since they do not assume the `JavaPlugin` is applied. 

What's interesting is that this is seemingly the first code in this class which makes any assumptions about there being production code at all (which sort-of makes sense). You don't necessarily _need_ production sources to run tests. Maybe someone could just want to run some arbitrary tests on a JVM, using a well-established testing framework, but to test something else which is not production code compiled in that project. 

This leads me to believe that it may not be the most correct for this class to be aware of production sources as well, and it seems like any logic which interacts with production should could live either in the `JavaPlugin` or a plugin which requires the `JavaPlugin` to be applied. 

Wall of text here, sorry. But I think this brings up an interesting point. I'm not really proposing any better solution right now, and I'm fine with this merging for now, though I'm interested in discussing this further.

Reply:
> Maybe someone could just want to run some arbitrary tests on a JVM, using a well-established testing framework, but to test something else which is not production code compiled in that project.

I think we definitely should support this case.

> This leads me to believe that it may not be the most correct for this class to be aware of production sources as well, and it seems like any logic which interacts with production should could live either in the JavaPlugin or a plugin which requires the JavaPlugin to be applied.

I agree with this, too.